packages/terra-popup/src/_PopupContent.jsx

```diff
@@ -0,0 +1,286 @@
+    const header = <div className="terra-PopupContent-header">{button}</div>;
+    return <ContentContainer header={header} fill>{children}</ContentContainer>;
+  }
+
+  static isFullScreen(height, maxHeight, width, maxWidth) {
+    if (maxHeight <= 0 || maxWidth <= 0) {
+      return false;
+    }
+    return height >= maxHeight && width >= maxWidth;
+  }
+
+  static isMarginValid(dimension, dimensionMax) {
+    if (dimensionMax <= 0) {
+      return true;
+    }
+    return dimensionMax >= dimension + POPUP_MARGIN;
+  }
+
+  static shouldShowArrow(arrow, arrowPosition, contentHeight, contentHeightMax, contentWidth, contentWidthMax) {
+    if (arrow) {
+      if (['top', 'bottom'].indexOf(arrowPosition) >= 0) {
+        return PopupContent.isMarginValid(contentHeight, contentHeightMax);
+      }
+      return PopupContent.isMarginValid(contentWidth, contentWidthMax);
+    }
+    return false;
+  }
+
+  constructor(props) {
+    super(props);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.handleKeydown = this.handleKeydown.bind(this);
+    this.handleResize = this.debounce(this.handleResize.bind(this), 100);
+  }
+
+  componentDidMount() {
+    if (this.props.closeOnEsc) {
+      document.addEventListener('keydown', this.handleKeydown);
+    }
+    if (this.props.closeOnResize) {
+      window.addEventListener('resize', this.handleResize);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.closeOnEsc) {
+      document.removeEventListener('keydown', this.handleKeydown);
+    }
+
+    if (this.props.closeOnResize) {
+      window.removeEventListener('resize', this.handleResize);
+    }
+  }
+
+  handleResize(event) {
+    if (this.props.closeOnResize && this.props.onRequestClose) {
+      this.props.onRequestClose(event);
+    }
+  }
+
+  handleClickOutside(event) {
+    if (this.props.closeOnOutsideClick && this.props.onRequestClose) {
+      this.props.onRequestClose(event);
+    }
+  }
+
+  handleKeydown(event) {
+    if (event.keyCode === KEYCODES.ESCAPE && this.props.onRequestClose) {
+      this.props.onRequestClose(event);
+      event.preventDefault();
+    }
+  }
+
+  debounce(fn, delay) {
+    let timer = null;
+    return (...args) => {
+      const context = this;
+      clearTimeout(timer);
+      timer = setTimeout(() => {
+        fn.apply(context, args);
+      }, delay);
+    };
+  }
+
+  render() {
+    const {
+      arrow,
+      arrowPosition,
+      children,
+      classNameInner,
+      closeOnEsc,
+      closeOnOutsideClick,
+      closeOnResize,
+      contentHeight,
+      contentHeightMax,
+      contentWidth,
+      contentWidthMax,
+      isHeaderDisabled,
+      onRequestClose,
+      refCallback,
+      ...customProps
+    } = this.props;
+
+    const showArrow = PopupContent.shouldShowArrow(arrow, arrowPosition, contentHeight, contentHeightMax, contentWidth, contentWidthMax);
```

Comment:
It's more a style choice, if a method is purely functional, with no transformative behavior, I tend to prefer static methods.